src/client/components/network-import/cy3-network-import-dialog.js

```diff
@@ -0,0 +1,172 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { NetworkEditorController } from '../network-editor/controller';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogActions from '@material-ui/core/DialogActions';
+import Dialog from '@material-ui/core/Dialog';
+import RadioGroup from '@material-ui/core/RadioGroup';
+import Radio from '@material-ui/core/Radio';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Button from '@material-ui/core/Button';
+import { Alert, AlertTitle } from '@material-ui/lab';
+import Link from '@material-ui/core/Link';
+import CheckCircleIcon from '@material-ui/icons/CheckCircle';
+import CancelIcon from '@material-ui/icons/Cancel';
+
+const CY3_URL = 'http://localhost:1234';
+
+export class Cy3NetworkImportDialog extends Component {
+
+  constructor(props) {
+    super(props);
+    this.controller = props.controller;
+    this.onClose = props.onClose;
+    this.radioGroupRef = {};
+    this.state = {
+      data: null,
+      value: null,
+      error: null,
+    };
+  }
+
+  handleEntering() {
+    if (this.radioGroupRef.current != null) {
+      this.radioGroupRef.current.focus();
+    }
+  }
+
+  handleCancel() {
+    this.onClose();
+  }
+
+  handleOk() {
+    const netId = this.state.value;
+    this.onClose();
+
+    fetch(`${CY3_URL}/v1/networks/${netId}/views`)
```

Comment:
You could make this more manageable with async/await.

You  just need to mark `async handleOk()`, and then you can use `let response = await fetch(...)` etc. in a more linear, sequential style that looks a lot like synchronous code.

See also https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Async_await

Reply:
Cool! I've just pushed some changes. Let me know what you think. Thanks!